### DISCUSSION TITLE
Wallet addition workflow

### DISCUSSION THREAD

**Discussion by sakulstra** [aave/interface#626] [Ideas]
I think there should be a guideline on wallet additions, requirements for listing/ unlising.
In the aave/interface wallets not satisfying certain quality goals (smol bundle size, wide browser support) were not being migrated from aave/ui.

I think for the aave interface it's important to:
1. not pick sides/favorites
2. don't allow uncontrolled bundle size growth as it decreases ui/ux for all users

Therefore It might be a good idea to define rules which must be fulfilled to get a wallet listed on the app.
Wallet's that use injected provider or wallet-connect usually don't need any special integration so this is only really relevant for "other wallets".

Perhaps a reasonable requirements list would be:
- shouldn't increase bundle size by more than ~10kb (arbitrary chosen number that seems "small enough" to me)
- needs to support switching and adding networks via rpc calls
- needs to work with ssr (as otherwise it can't be bundled and dynamic imports don't work well with ipfs-gateway rate limits in mind)

**Comment by sendra** [comment]
I fully agree. The interface should try to be the least opinionated as possible. 
It is very dangerous to open the gates to any wallet integration, as apart from bloating the bundle size, it also can add complexity for users when trying to connect.

I would also try to add a certain minimums the wallets would need to have to be added, like:
- minimum number of MAU
-  spread of usage of the wallet in the ecosystem
